pkg/gardenlet/controller/managedseed/managedseed.go

```diff
@@ -1,189 +0,0 @@
-	GardenletDefaultKubeconfigBootstrapSecretName = "gardenlet-kubeconfig-bootstrap"
-)
-
-// Controller controls ManagedSeeds.
-type Controller struct {
-	log logr.Logger
-
-	gardenCache runtimecache.Cache
-	config      *config.GardenletConfiguration
-	reconciler  reconcile.Reconciler
-
-	managedSeedInformer runtimecache.Informer
-	seedInformer        runtimecache.Informer
-
-	managedSeedQueue workqueue.RateLimitingInterface
-
-	numberOfRunningWorkers int
-	workerCh               chan int
-}
-
-// NewManagedSeedController creates a new Gardener controller for ManagedSeeds.
-func NewManagedSeedController(
-	ctx context.Context,
-	log logr.Logger,
-	gardenCluster cluster.Cluster,
-	seedCluster cluster.Cluster,
-	shootClientMap clientmap.ClientMap,
-	config *config.GardenletConfiguration,
-	imageVector imagevector.ImageVector,
-) (
-	*Controller,
-	error,
-) {
-	log = log.WithName(ControllerName)
-
-	managedSeedInformer, err := gardenCluster.GetCache().GetInformer(ctx, &seedmanagementv1alpha1.ManagedSeed{})
-	if err != nil {
-		return nil, fmt.Errorf("could not get ManagedSeed informer: %w", err)
-	}
-
-	seedInformer, err := gardenCluster.GetCache().GetInformer(ctx, &gardencorev1beta1.Seed{})
-	if err != nil {
-		return nil, fmt.Errorf("could not get Seed informer: %w", err)
-	}
-
-	var (
-		valuesHelper = NewValuesHelper(config, imageVector)
-		actuator     = newActuator(
-			gardenCluster.GetConfig(),
-			gardenCluster.GetAPIReader(),
-			gardenCluster.GetClient(),
-			seedCluster.GetClient(),
-			shootClientMap,
-			valuesHelper,
-			gardenCluster.GetEventRecorderFor(ControllerName+"-controller"),
-		)
-	)
-
-	return &Controller{
-		log:                 log,
-		gardenCache:         gardenCluster.GetCache(),
-		config:              config,
-		reconciler:          newReconciler(gardenCluster.GetClient(), actuator, config.Controllers.ManagedSeed),
-		managedSeedInformer: managedSeedInformer,
-		seedInformer:        seedInformer,
-		managedSeedQueue:    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ManagedSeed"),
-		workerCh:            make(chan int),
-	}, nil
-}
-
-// Run runs the Controller until the given context is cancelled.
-func (c *Controller) Run(ctx context.Context, workers int) {
-	var waitGroup sync.WaitGroup
-
-	c.managedSeedInformer.AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controllerutils.ManagedSeedFilterFunc(ctx, c.gardenCache, confighelper.SeedNameFromSeedConfig(c.config.SeedConfig)),
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    c.managedSeedAdd,
-			UpdateFunc: c.managedSeedUpdate,
-			DeleteFunc: c.managedSeedDelete,
-		},
-	})
-
-	// Add event handler for controlled seeds
-	c.seedInformer.AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controllerutils.SeedOfManagedSeedFilterFunc(ctx, c.gardenCache, confighelper.SeedNameFromSeedConfig(c.config.SeedConfig)),
-		Handler: &kutil.ControlledResourceEventHandler{
```

Comment:
Now it is time to drop this `kutil.ControlledResourceEventHandler` and everything related to it.